test/sharness/lib/test-lib.sh

```diff
@@ -41,6 +41,43 @@ SHARNESS_LIB="lib/sharness/sharness.sh"
 
 # Please put go-ipfs specific shell functions below
 
+if test "$GOIPFSTEST_ENSURE_CLEAN_START" = 1 ; then
```

Comment:
`make test_sharness_expensive` actually invokes sharness tests in parallel, IIRC. I don't think this logic will work with that.

_However_, when running a sharness test, we `cd` into a "trash" directory. We _could_ kill off all IPFS instances running in that trash directory.

Reply:
Good idea! I definitely wouldn't want to kill them, but rather detect them and fail. The main point of this eercise was to stop my workstation being filled up with random not-shutdown-daemons